File: src/cp1/src/cp1.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import Int32
import time

def talker():
    pub = rospy.Publisher('talk', Int32, queue_size=10)
    data=Int32 (input('please input a number: '))
    pub.publish(data)
    time.sleep(0.5)

def callback(msg):
    print('return value is: ' + str(msg.data))

def main():
    rospy.init_node('talker', anonymous=True)
    rospy.Subscriber('listen',Int32, callback)
    try:
        while (True):
            #print('hello')
            talker()
    except rospy.ROSInterruptException:
        pass
main()

File: src/cp1/src/cp1.cpp
#include <ros/ros.h>
#include <std_msgs/Int32.h>

ros::Publisher num_pub;
bool arduino_return_state = true;

void num_cb(const std_msgs::Int32::ConstPtr& ptr){
  ROS_INFO_STREAM("Num received from Arduino is: " << ptr->data);
  arduino_return_state = true;
}

int main(int argc, char** argv){
  ros::init(argc, argv, "cp1");
  ros::NodeHandle nh("");
  std_msgs::Int32 num;

  num_pub = nh.advertise<std_msgs::Int32>("talk", 10);
  ros::Subscriber num_sub = nh.subscribe("listen", 10, &num_cb);

  try
  {
    ROS_INFO("[Check Point 1]: Initializing node");

    while(ros::ok){
      if (arduino_return_state){
        ROS_INFO_STREAM("Waiting user input...");
        std::cin >> num.data;
        ROS_INFO_STREAM("Num sent to Arduino is: " << num.data);
        num_pub.publish(num);
        arduino_return_state = false;
      }
      ros::spinOnce();
    }
      
  }
  catch (const char* s)
  {
    ROS_FATAL_STREAM("[Check Point 1]: " << s);
  }
  catch (...)
  {
    ROS_FATAL_STREAM("[Check Point 1]: Unexpected error");
  }

  return 0;
}
